_playwright-tests/UI/TemplateCRUD.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories, navigateToTemplates } from './helpers/navHelpers';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+import { deleteAllTemplates } from './helpers/deleteTemplates';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+
+const templateNamePrefix = 'template_CRUD';
+const repoNamePrefix = 'custom_repo-template';
+
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+export const templateName = `${templateNamePrefix}-${randomName()}`;
+
+const smallRHRepo = 'Red Hat CodeReady Linux Builder for RHEL 9 ARM 64 (RPMs)';
+
+test.describe('Templates CRUD', () => {
+  test('Add, Read, update, delete a template', async ({ page }) => {
+    await test.step('Delete any templates and template test repos that exist', async () => {
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+      await deleteAllTemplates(page, `&search=${templateNamePrefix}`);
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+    });
+    await test.step('Create a repository', async () => {
```

Comment:
this whole step could be just an API call and a poll for the status 😅 
would help with speeding up our test suite and eliminate an extra navigation which are slow 🐌 
_(i.e.: no navigation to repos, straight to templates)_

Reply:
@dominikvagner , I have changed that section and added a helper, please review it.